src/qibolab/sequence.py

```diff
@@ -93,17 +94,36 @@ def pulse_channels(self, pulse_id: int) -> list[ChannelId]:
         """Find channels on which a pulse with a given id plays."""
         return [channel for channel, pulse in self if pulse.id == pulse_id]
 
-    def concatenate(self, other: "PulseSequence") -> None:
+    def concatenate(self, other: Iterable[_Element]) -> None:
```

Comment:
Why do we accept any iterable here? Isn't `PulseSequence` sufficient? Same for `__or__` and `__ior__`.

Reply:
https://github.com/qiboteam/qibolab/blob/746fe8f3eba84ae869d23927851a4661f5443ff7/tests/integration/test_sequence.py#L22-L29

The explicit pulse-like here (the `Delay`) is contained in a `list`, which is perfectly homogeneous to a `PulseSequence` (since `PulseSequence` is a `list` subclass), but it's not a `PulseSequence`.

In general, it is very cheap to support, and quite user-friendly (e.g. even an iterator over `_Element` would work, so you can use literals such as list comprehension and generators, that otherwise should be manually wrapped in a `PulseSequence`).